### DISCUSSION TITLE
使用natter映射v2ray端口，一段时间后代理连接失败，natter日志显示fwd-socket: cannot forward port: Too many threads

### DISCUSSION THREAD

**Discussion by mrbyfwg** [MikeWang000000/Natter#104] [Q&A]
重启natter后恢复正常
请问可能是什么原因
![图片](https://github.com/user-attachments/assets/80064c8a-f79f-43eb-92d4-5a0bdff0b4e5)

**Comment by MikeWang000000** [comment]
`-m socket` 模式（默认的模式）不适用于高并发连接的场景。因为它对于每个连接，都会建立一对线程用于收发。

如果你在路由器等资源有限的设备上运行，此时遭遇了外部高并发的攻击，这种场景下会建立过多线程，把你的设备打死。
Natter 为做保护，强制设定了线程数上限。

使用 `-m iptables` 模式就不存在这种限制，多少并发数都可以。
